src/gridfs/file.rs

```diff
@@ -246,10 +246,10 @@ impl File {
 
         let document =
             doc! {
-            "_id" => (try!(oid::ObjectId::new())),
-            "files_id" => (self.doc.id.clone()),
-            "n" => n,
-            "data" => (BinarySubtype::Generic, vec_buf)
+            "_id": try!(oid::ObjectId::new()),
```

Comment:
Unrelated: do we want to do a mass switch from `try!` to `?` at some point? The only downside would be breaking things on older compilers, but I don't get the sense that Rust users generally stick to old stable versions

Reply:
SGTM. I'd like to do it in a more selective/incremental manner if possible — I know there have been cases of unintentional slowdowns making that change due to objects with result-wrapper semantics (i.e. they wrap results and basically have two values based on ok/err values), and I can't quite remember if we have anything like that in our codebase.